### PR TITLE
Added filtering support through translation layer.

### PR DESCRIPTION
**Pull request by AbhitejJohn**
Incudes changes from @genlu  as well.

### CODE REVIEW

**Comment by AbhitejJohn** [comment]
@dotnet-bot test this please.

**Comment by AbhitejJohn** [comment]
Here is the RFC: https://github.com/Microsoft/vstest-docs/pull/73

File: test/vstest.console.UnitTests/TestPlatformHelpers/TestRequestManagerTests.cs [hunk 1]
```diff
@@ -342,6 +342,60 @@ public void RunTestsWithSourcesShouldCallTestPlatformAndSucceed()
         }
 
         [TestMethod]
+        public void RunTestsWithSourceAndFilterInDesignModeShouldUseFilterFromPayload()
+        {
+            string testCaseFilterFromConsole = "TestFilterFromConsole";
```
Comment (smadala) [inline test/vstest.console.UnitTests/TestPlatformHelpers/TestRequestManagerTests.cs]: DRY.

**Review by smadala — APPROVED** [review]

**Comment by genlu** [comment]
FYI @KevinH-MS @peterwald

**Comment by peterwald** [comment]
LGTM :+1:

**Comment by AbhitejJohn** [comment]
@dotnet-bot test this please.

**Comment by AbhitejJohn** [comment]
The PR build for this branch does not seem to be working. Triggered one [here](https://github.com/Microsoft/vstest/pull/1052) on master to ensure tests are green.
@genlu: Would you be able to validate E2E before we go ahead and push this in?

**Comment by genlu** [comment]
@AbhitejJohn Yes, I did some quick test in VS using the new API, everything works fine.

**Review by singhsarab — APPROVED** [review]
LGTM. Assuming you have validated all the scenarios with respect to protocol versions.

**Comment by AbhitejJohn** [comment]
This isn't required for 15.4 any more. Will work on editing this appropriately for 15.5.

**Pull request closed without merging**